Generic In-Memmory/test/bg/sofia/uni/fmi/mjt/cache/MemCacheTest.java

```diff
@@ -0,0 +1,193 @@
+    }
+
+    @Test
+    public void testShouldNotPutValueToTheCache() throws CapacityExceededException {
+        cacheWithLongCapacity.set(null, 1, null);
+        assertEquals(0, cacheWithLongCapacity.size());
+        cacheWithLongCapacity.set("", null, null);
+        assertEquals(0, cacheWithLongCapacity.size());
+    }
+
+    @Test
+    public void testShouldPutValueToTheCache() throws CapacityExceededException {
+        cacheWithLongCapacity.set("First", 1, null);
+        assertEquals(1, cacheWithLongCapacity.size());
+        cacheWithLongCapacity.set("Second", 2, null);
+        assertEquals(2, cacheWithLongCapacity.size());
+    }
+
+    @Test
+    public void testShouldGetAllItems() throws CapacityExceededException {
+        cacheWithLongCapacity.set("First", 1, null);
+        cacheWithLongCapacity.set("Second", 1, LocalDateTime.now().minusDays(2));
+        cacheWithLongCapacity.set("Third", 1, LocalDateTime.now().minusDays(123123));
+
+        assertEquals(3, cacheWithLongCapacity.size());
+    }
+
+    @Test
+    public void testShouldGetExpirationDate() throws CapacityExceededException {
+        LocalDateTime time = LocalDateTime.now().plusDays(1);
+        cacheWithLongCapacity.set("First", 1, time);
+        assertEquals(time, cacheWithLongCapacity.getExpiration("First"));
+    }
+
+    @Test
+    public void testShouldGetNullExpirationDateWhenElementNotExist() {
+        assertNull(cacheWithLongCapacity.getExpiration("First"));
+    }
+
+    @Test
+    public void testShouldReplaceValueAndTime() throws CapacityExceededException {
+        LocalDateTime time = LocalDateTime.now().plusDays(1);
+
+        cacheWithLongCapacity.set("First", 1, null);
+        assertEquals(1, cacheWithLongCapacity.get("First").intValue());
+        assertNull(cacheWithLongCapacity.getExpiration("First"));
+
+        cacheWithLongCapacity.set("First", 4, time);
+        assertEquals(4, cacheWithLongCapacity.get("First").intValue());
+        assertEquals(time, cacheWithLongCapacity.getExpiration("First"));
+
+        assertEquals(1, cacheWithLongCapacity.size());
+    }
+
+    @Test
+    public void testShouldRemoveExpiredItemWhenGetIt() throws CapacityExceededException {
+        LocalDateTime time = LocalDateTime.now().minusDays(1);
+
+        cacheWithLongCapacity.set("First", 1, time);
+
+        Integer result = cacheWithLongCapacity.get("First");
+        assertEquals(0, cacheWithLongCapacity.size());
+        assertNull(result);
+    }
+
+    @Test
+    public void testShouldRemoveExpiredItemWhenAddingNewWithExpiredCapacity() throws CapacityExceededException {
+        LocalDateTime notExpiredTime = LocalDateTime.now().plusDays(1);
+        LocalDateTime expiredTime = LocalDateTime.now().minusDays(1);
+
+        cacheWithLongCapacity.set("First", 1, notExpiredTime);
+        cacheWithLongCapacity.set("Second", 2, null);
+        cacheWithLongCapacity.set("Third", 3, expiredTime);
+        cacheWithLongCapacity.set("Fourth", 4, null);
```

Comment:
Assert size before and after get operation to ensure that it actually removed one item from the cache